feat: Build of the crash reporter is now don in a docker container to ensure compatibility with met system

Angel-Padilla opened the pull request:
- Add a container to build for an older version of GLIBC
    The build will now run in a debian:bookworm docker image, so that the binary built will use the GLIBC_2.36 instead of the current 2.38 version required.

- fix: change ENV variables read to the ones passed by the failed service
    According to the [systemd exec manual](https://www.freedesktop.org/software/systemd/man/252/systemd.exec.html), the variables passed to the service called onFailure are
```
MONITOR_UNIT
MONITOR_SERVICE_RESULT
MONITOR_EXIT_CODE
MONITOR_EXIT_STATUS
MONITOR_INVOCATION_ID

since version 251, we are running on version 252.
```
- feat: change hardcoded cd command
    change hardcoded `/systemd-crash-reporter` to dinamically get the path of the running script, so it can be run in or outside of the container

This commit requires approval of the PR: [Fix crash reporter issues #37 ](https://github.com/MeticulousHome/meticulous-machine/pull/37) on the `meticulous-machine` repo

Angel-Padilla committed:
feat:
The build will now run in a debian:bookworm docker image, so that the
binary built will use the GLIBC_2.36 instead of the current 2.38 version
required.

Angel-Padilla committed:
fix: change ENV variables read to the ones passed by the failed service

According to the systemd.exec documentation, the variables passed to the
service called onFailure are

MONITOR_UNIT
MONITOR_SERVICE_RESULT
MONITOR_EXIT_CODE
MONITOR_EXIT_STATUS
MONITOR_INVOCATION_ID

since version 251, we are running on version 252.

[systemd manual](https://www.freedesktop.org/software/systemd/man/252/systemd.exec.html)

Mimoja commented on build.sh:
```diff
@@ -1,23 +1,23 @@
 fi
 export TARGET_CC=aarch64-linux-gnu-gcc
 
+cd "/systemd-crash-reporter"
```
If you don't do this but use the pwd of this script we can still build it locally without a docker file around it and with reduced privileges

Angel-Padilla replied:
You mean something like
``` sh
DIR=$(dirname "$(realpath $0)")
cd "$DIR"
```
?

I didn't think about it. But I dont know why would we want to run it locally if we still need the downgraded GLIBC version for it to run on the machine

Mimoja replied:
Yes. We are doing it in multiple locations already

Angel-Padilla committed:
feat: change hardcoded cd command

change hardcoded `/systemd-crash-reporter` to dinamically get the path
of the running script, so it can be run in or outside of the container

Mimoja commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# ENTRYPOINT ["systemd-crash-reporter/build.sh"]
```
This feels redundant. You could use bash or not explicit entry point?

Angel-Padilla replied:
Sorry, forgot to delete it

Angel-Padilla committed:
chore: remove commented code
